### PR TITLE
Oxidize utils::run_interactively

### PR DESCRIPTION
**Pull request by rnestler**
Finishes #334 :D

### CODE REVIEW

**Comment by Minoru** [comment]
BTW, I'm going AFK for the weekend, so don't take it wrong if I don't reply before Monday.

File: rust/libnewsboat-ffi/src/utils.rs [hunk 1]
```diff
@@ -323,6 +323,18 @@ pub unsafe extern "C" fn rs_make_title(input: *const c_char) -> *mut c_char {
     })
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn rs_run_interactively(
+    command: *const c_char,
+    caller: *const c_char,
+) -> i32 {
+    abort_on_panic(|| {
+        let rs_command = CStr::from_ptr(command).to_string_lossy();
```
[the remark this answers is not in the document]
Comment (rnestler) [reply rust/libnewsboat-ffi/src/utils.rs]: Fixed.

File: rust/libnewsboat/src/utils.rs [hunk 2]
```diff
@@ -527,6 +527,24 @@ pub fn make_title(rs_str: String) -> String {
     }
 }
 
+pub fn run_interactively(command: &str, caller: &str) -> i32 {
+    log!(Level::Debug, &format!("{}: running `{}'", caller, command));
+    Command::new("sh")
+        .arg("-c")
+        .arg(command)
```
[the remark this answers is not in the document]
Comment (rnestler) [reply rust/libnewsboat/src/utils.rs]: Ah, thanks for clearing this up :slightly_smiling_face:

**Comment by rnestler** [comment]
> BTW, I'm going AFK for the weekend, so don't take it wrong if I don't reply before Monday.

No problem, I hope you enjoyed your weekend :slightly_smiling_face:

**Comment by Minoru** [comment]
Thanks for an update! So what about tests? Do you see any way to check that this actually works?

**Comment by Minoru** [comment]
@rnestler, ping?

**Comment by rnestler** [comment]
> Thanks for an update! So what about tests? Do you see any way to check that this actually works?

Sorry, I somehow forgot about this open PR during past week :see_no_evil: 

I added a few Rust tests which seem to work. I guess I can just add them in C++ as well.

**Pull request merged**